### PR TITLE
Add disconnect option when changing databases

### LINKED ISSUE #904: Suggestion: Add Disconnect to status bar server connection shortcut
- MSSQL Extension Version: 1.0.0
- VSCode Version: 1.13.1
- OS Version: any

If you are connected to a sql server and click on the server info text in the status bar, that opens the "choose database" dialog.

Is it possible to change that behavior so you have the option to either choose a database or just disconnect?

### PR DESCRIPTION
**Pull request by MattIrv**
Closes #904 

This change adds an option to disconnect when clicking on the connection status in the bottom-right, as an alternative to changing the connected database. If you choose to disconnect, it will prompt for a confirmation before disconnecting.

### CODE REVIEW

File: src/views/connectionUI.ts [hunk 1]
```diff
@@ -195,21 +195,49 @@ export class ConnectionUI {
...
                 placeHolder: LocalizedConstants.msgChooseDatabasePlaceholder
             };
 
             // show database picklist, and modify the current connection to switch the active database
-            self.vscodeWrapper.showQuickPick<Interfaces.IConnectionCredentialsQuickPickItem>(pickListItems, pickListOptions).then( selection => {
+            self.vscodeWrapper.showQuickPick<vscode.QuickPickItem>(pickListItems, pickListOptions).then( selection => {
+                if (selection === disconnectItem) {
+                    self.handleDisconnectChoice().then(() => resolve(undefined), err => reject(err));
+                }
                 if (typeof selection !== 'undefined') {
```
Comment (kevcunnane) [inline src/views/connectionUI.ts]: Shouldn't this be an else if? While resolve will eventually return from the promise it doesn't actually block the subsequent code from being run I think?

File: src/views/connectionUI.ts [hunk 2]
```diff
@@ -195,21 +195,49 @@ export class ConnectionUI {
...
+
             const pickListOptions: vscode.QuickPickOptions = {
                 placeHolder: LocalizedConstants.msgChooseDatabasePlaceholder
             };
 
             // show database picklist, and modify the current connection to switch the active database
-            self.vscodeWrapper.showQuickPick<Interfaces.IConnectionCredentialsQuickPickItem>(pickListItems, pickListOptions).then( selection => {
+            self.vscodeWrapper.showQuickPick<vscode.QuickPickItem>(pickListItems, pickListOptions).then( selection => {
+                if (selection === disconnectItem) {
+                    self.handleDisconnectChoice().then(() => resolve(undefined), err => reject(err));
```
Comment (kevcunnane) [inline src/views/connectionUI.ts]: At this point if you reject what happens? I think it's possible you should handle it by showing the error to the user.

Comment (MattIrv) [inline src/views/connectionUI.ts]: Turns out that rejecting here does show the error to the user

**Comment by MattIrv** [comment]
Received in person sign-off from Kevin

**Pull request merged**